sewer/dns_providers/auroradns.py

```diff
@@ -0,0 +1,80 @@
+# DNS Provider for AuroRa DNS from the dutch hosting provider pcextreme
+# https://www.pcextreme.nl/aurora/dns
+# Aurora uses libcloud from apache
+# https://libcloud.apache.org/
+
+from libcloud.dns.providers import get_driver
+from libcloud.dns.types import Provider, RecordType
+import tldextract
+from structlog import get_logger
+from . import common
+
+
+class AuroraDns(common.BaseDns):
+    """
+    """
+
+    def __init__(self, AURORA_API_KEY, AURORA_SECRET_KEY):
+
+        self.AURORA_API_KEY = AURORA_API_KEY
+        self.AURORA_SECRET_KEY = AURORA_SECRET_KEY
+        self.dns_provider_name = 'aurora'
+
+        self.logger = get_logger(__name__).bind(
+            dns_provider_name=self.dns_provider_name)
+
+    def create_dns_record(self, domain_name, base64_of_acme_keyauthorization):
+        self.logger.info('create_dns_record')
+
+        # delete any prior existing DNS authorizations that may exist already
+        self.delete_dns_record(
+            domain_name=domain_name,
+            base64_of_acme_keyauthorization=base64_of_acme_keyauthorization)
+
+        extractedDomain = tldextract.extract(domain_name)
```

Comment:
@komuW. The need for splitting up the domain is as follows.

When having test1.example.com
* example.com is used to select the right domain
* test1 is used to select or add the record

It is not possible to select, create or delete a record based on the whole domain name (test.example.com). This is the reason why it is needed tot split up the record.

Splitting up records by hand or regular expression is difficult, because you can have example.com and example.co.uk. So you can't select the part's (tld, domain name, sub domains) based on the . in de fqdn. tldextract uses the Public Suffix List (https://publicsuffix.org/learn/) to solve this problem. 

There are some alternatives posible, wich I would no prefer:
* Asking user to give sub and domain part. This way there is no processing needed. 
* Include a tldextract like function with sewer.

I have looked before at urlparse. Urlparse kan selects parts of a url, but not parts of a domain. So this is not suitable.